pkg/vpcmodel/output.go

```diff
@@ -216,17 +218,23 @@ func (of *serialOutputFormatter) WriteOutput(cConfigs *MultipleVPCConfigs, conns
 		}
 		return of.AggregateVPCsOutput(outputPerVPC, uc, outFile)
 	}
-	for uid, vpcConfig := range cConfigs.Configs() {
-		vpcAnalysisOutput, err2 :=
-			of.createSingleVpcFormatter().WriteOutput(vpcConfig, cConfigs.ConfigToCompare(uid), conns[uid], subnetsConns[uid],
-				subnetsDiff, "", grouping, uc, explainStruct)
-		if err2 != nil {
-			return "", err2
-		}
-		// its diff or explain mode, we have only one vpc on each map:
-		return of.WriteDiffOrExplainOutput(vpcAnalysisOutput, uc, outFile)
+	// singleVPCAnalysis: either diff or explain. In either case conn and subnet conn are non-relevant, thus passing nil
+	// diff compares between two single vpc configs
+	// explain works on a specific config, either single or multiple; the relevant config for explain is kept
+	// in its structs, thus the configs passed here are non-relevant for it; the flow is such that valid main config must be passed,
+	// also for explain (even though it does not affect the output)
+	var toCompareConfig *VPCConfig
+	if uc == EndpointsDiff || uc == SubnetsDiff {
+		toCompareConfig = cConfigs.aConfigToCompare()
+	}
+	vpcAnalysisOutput, err2 :=
```

Comment:
change `err2` -> `err` throughout this function.

Reply:
done

Reply:
I still see `err2` in this function

Reply:
fixed